### PR TITLE
Fix more type issues

### PR DESCRIPTION
**Pull request by Gobot1234**

### CODE REVIEW

File: taskgroup/taskgroups.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ async def _cmgr_factory(self) -> AsyncGenerator[Self, None]:
                     # cycles (bad for GC); let's not keep a reference to
                     # a bunch of them.
                     errors = self._errors
-                    self._errors = None
+                    self._errors.clear()
```
Comment (graingert) [inline taskgroup/taskgroups.py]: I'd rather keep this per the original code

**Comment by Gobot1234** [comment]
@graingert pyright passes now

**Comment by graingert** [comment]
It looks like CI didn't run though?

**Comment by Gobot1234** [comment]
I'm not really sure why it's not on the pull request but it's on my branch https://github.com/Gobot1234/taskgroup/actions/runs/6379072600/job/17310832061

**Comment by graingert** [comment]
It's only `on: [push]`

**Review by graingert — APPROVED** [review]

**Pull request merged**